sql/core/src/main/scala/org/apache/spark/sql/internal/BaseSessionStateBuilder.scala

```diff
@@ -309,13 +309,14 @@ private[sql] trait WithTestConf { self: BaseSessionStateBuilder =>
   def overrideConfs: Map[String, String]
 
   override protected lazy val conf: SQLConf = {
+    val overrideConfigurations = overrideConfs
     val conf = parentState.map(_.conf.clone()).getOrElse {
       new SQLConf {
         clear()
         override def clear(): Unit = {
           super.clear()
           // Make sure we start with the default test configs even after clear
-          overrideConfs.foreach { case (key, value) => setConfString(key, value) }
+          overrideConfigurations.foreach { case (key, value) => setConfString(key, value) }
```

Comment:
Technically, this is touching `src/main` instead of `src/test`.

Reply:
maybe we should move this trait to `src/test`?

Reply:
@cloud-fan agree.

Reply:
I tried but failed. 
Warning from IDE: Interface `org.apache.spark.sql.internal.WithTestConf`, referenced in file `TestHive.scala`, will not be accessible from module `spark-hive`